### PR TITLE
planner: fix wrong argument for split aggregation (#20474)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #20474 to release-4.0

---

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/20254

Problem Summary:

Error reported when compiling a query.

### What is changed and how it works?

What's Changed:
If the argument of `count(distinct)` is not a column, use the scalar function as the argument of the split aggregation, instead of the projected column.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

N/A

### Release note 

- fix wrong argument for split aggregation

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -1332,6 +1332,106 @@ func (s *testIntegrationSuite) TestIssue16935(c *C) {
 	tk.MustQuery("SELECT * FROM t0 LEFT JOIN v0 ON TRUE WHERE v0.c0 IS NULL;")
 }
 
+<<<<<<< HEAD
+=======
```
Comment (qw4990) [inline planner/core/integration_test.go:1336]: Please resolve conflicts @eurekaka

**Review by winoros — COMMENTED** [review]
lgtm

**Review by zz-jason — APPROVED** [review]
LGTM

**Pull request merged**